neptune-core/src/application/util_proof/sent/mod.rs

```diff
@@ -0,0 +1,329 @@
+        /* pasted from <../reserves>
+        ============== */
+        // _
+        push {FIRST_NON_DETERMINISTICALLY_INITIALIZED_MEMORY_ADDRESS}
+        {&rustfield_utxo}
+        {&Utxo::get_field("lock_script_hash")}
+        read_mem 5
+        pop 1
+        // [lock_script_digest]
+        write_io 5
+        // _
+
+        push {FIRST_NON_DETERMINISTICALLY_INITIALIZED_MEMORY_ADDRESS}
+        // *w
+        {&WitnessMemory::get_field("aocl")}
+        // *aocl
+
+        // prepare the value for '7.'
+        dup 0
+        {&MmrAccumulator::get_field("peaks")}
+        // *aocl *aocl_peaks
+
+        // read `receiver_digest`
+        read_io 5
+        // *aocl *aocl_peaks [receiver_digest]
+
+        push {FIRST_NON_DETERMINISTICALLY_INITIALIZED_MEMORY_ADDRESS}
+        // _ *w
+        {&rustfield_senderrandomness} // omits `{&rustfield_membershipproof}` from <../reserves>
+        read_mem {Digest::LEN}
+        pop 1
+        // *aocl *aocl_peaks [receiver_digest] [sender_randomness]
+
+        /* ## this segment diverges from the pasted code
+        *it only affects the output sequence* */
+        push 0
+        push 0
+        push 0
+        push 0
+        push 0
+        dup 9
+        dup 9
+        dup 9
+        dup 9
+        dup 9
+        hash
+        write_io 5
+
+        push {FIRST_NON_DETERMINISTICALLY_INITIALIZED_MEMORY_ADDRESS}
+        // _ *w
+        {&rustfield_utxo}
+        // *aocl *aocl_peaks [receiver_digest] [sender_randomness] *utxo
+    };
+
+    let main = triton_asm! {
+        read_mem 1
+        addi 2
+        // _ utxo_size *utxo+1
+        swap 1
+        // _ *utxo+1 utxo_size
+        call {lib_hash_varlen}
+        hint utxo_hash = stack[0..5]
+        // *aocl *aocl_peaks [receiver_digest] [sender_randomness] [utxo_hash]
+
+        push {FIRST_NON_DETERMINISTICALLY_INITIALIZED_MEMORY_ADDRESS}
+        {&rustfield_utxodigest}
+        read_mem 5
+        assert_vector
+        // *aocl *aocl_peaks [receiver_digest] [sender_randomness] [utxo_hash]
+
+        call {lib_ms_commit}
+        // *aocl *aocl_peaks [canonical_commitment]
+
+        /* 7. */
+        dup 6 {&MmrAccumulator::get_field("leaf_count")}
+        // *aocl *aocl_peaks [canonical_commitment] *num_leafs
+
+        addi 1 read_mem {u64_stack_size} pop 1
+        // *aocl *aocl_peaks [canonical_commitment] [num_leafs]
+
+        push {FIRST_NON_DETERMINISTICALLY_INITIALIZED_MEMORY_ADDRESS}
+        {&rustfield_aoclleafindex} // omits `{&rustfield_membershipproof}` from <../reserves>
+        read_mem {u64_stack_size}
+        pop 1
+        // *aocl *aocl_peaks [canonical_commitment] [num_leafs] [aocl_leaf_index]
+
+        push {FIRST_NON_DETERMINISTICALLY_INITIALIZED_MEMORY_ADDRESS}
+        {&WitnessMemory::get_field("auth_path_aocl")}
+        {&WitnessMemory::get_field("authentication_path")}
+        // *aocl *aocl_peaks [canonical_commitment] [num_leafs] [aocl_leaf_index] *auth_path
+        swap 2
```

Comment:
This is not how `swap 2` works.

The stack `_  [num_leafs] [aocl_leaf_index] *auth_path` is short-hand for `_ num_leafs_hi num_leafs_lo aocl_leaf_index_hi aocl_leaf_index_lo *auth_path`.

What you want is `place 4`.